incubator/hnc/api/v1alpha1/hierarchy_types.go

```diff
@@ -28,6 +28,7 @@ const (
 	CritAncestor              Code = "CRIT_ANCESTOR"
 	ObjectOverridden          Code = "OBJECT_OVERRIDDEN"
 	ObjectDescendantOverriden Code = "OBJECT_DESCENDANT_OVERRIDDEN"
+	ObjectHasFinalizer        Code = "OBJECT_HAS_FINALIZER"
```

Comment:
We can probably rename the condition to ```ObjectInternal```, considering we are checking 3 fields for now ```DeletionGracePeriodSeconds```, ```DeletionTimestamp```,```Finalizers```. We may also check more object internal issues in the future in the ```ensureCanPropagate()``` function suggested by Adrian below.

Reply:
Done!